File: app/src/main/java/ch/bailu/aat/util/ui/UiThemeLightHeader.java
package ch.bailu.aat.util.ui;

import android.graphics.Color;
import android.graphics.Typeface;
import android.view.View;
import android.widget.TextView;

public class UiThemeLightHeader  extends UiThemeLight {

    public final static int BUTTON_GRAY = Color.GRAY;

    private final int hl_color;

    public UiThemeLightHeader(int hl_color) {
        this.hl_color = hl_color;
    }

    @Override
    public void topic(TextView v) {
        v.setTextColor(hl_color);
        v.setTextSize(HEADER_TEXT_SIZE *1.5f);
        v.setTypeface(null, Typeface.BOLD);
    }

    @Override
    public void header(TextView v) {
        v.setTextColor(hl_color);
        v.setTextSize(HEADER_TEXT_SIZE);
        v.setTypeface(null, Typeface.BOLD);
    }

    @Override
    public int getHighlightColor() {
        return hl_color;
    }


    @Override
    public void button(View v) {
        v.setBackgroundDrawable(AppTheme.getButtonDrawable(BUTTON_LIGHT_GRAY, BUTTON_GRAY));
    }
}

File: aat-android/src/main/java/ch/bailu/aat/util/ui/theme/UiThemeLightHeader.kt
package ch.bailu.aat.util.ui.theme

import android.graphics.Color
import android.graphics.Typeface
import android.view.View
import android.widget.TextView

class UiThemeLightHeader(private val hlColor: Int) : UiThemeLight() {
    override fun topic(v: TextView) {
        v.setTextColor(hlColor)
        v.textSize = UiTheme.HEADER_TEXT_SIZE * 1.5f
        v.setTypeface(null, Typeface.BOLD)
    }

    override fun header(v: TextView) {
        v.setTextColor(hlColor)
        v.textSize = UiTheme.HEADER_TEXT_SIZE
        v.setTypeface(null, Typeface.BOLD)
    }

    override fun getHighlightColor(): Int {
        return hlColor
    }

    override fun button(v: View) {
        v.setBackgroundDrawable(AppTheme.getButtonDrawable(BUTTON_LIGHT_GRAY, BUTTON_GRAY))
    }

    companion object {
        const val BUTTON_GRAY = Color.GRAY
    }
}
